src/include/mallocMC/distributionPolicies/XMallocSIMD_impl.hpp

```diff
@@ -76,19 +75,19 @@ namespace DistributionPolicies{
  * default-struct < template-struct < command-line parameter
  */
 #ifndef MALLOCMC_DP_XMALLOCSIMD_PAGESIZE
-#define MALLOCMC_DP_XMALLOCSIMD_PAGESIZE Properties::pagesize::value
+#define MALLOCMC_DP_XMALLOCSIMD_PAGESIZE (Properties::pagesize)
 #endif
-      BOOST_STATIC_CONSTEXPR uint32 pagesize      = MALLOCMC_DP_XMALLOCSIMD_PAGESIZE;
+      static constexpr uint32 pagesize      = MALLOCMC_DP_XMALLOCSIMD_PAGESIZE;
 
       //all the properties must be unsigned integers > 0
-      BOOST_STATIC_ASSERT(!std::numeric_limits<typename Properties::pagesize::type>::is_signed);
+      static_assert(!std::numeric_limits<decltype(Properties::pagesize)>::is_signed, ""); // TODO(bgruber): I think this is too strict. std::is_convertible_v should be enough
 
       // \TODO: The static_cast can be removed once the minimal dependencies of
       //        this project is are at least CUDA 7.0 and gcc 4.8.2
-      BOOST_STATIC_ASSERT(static_cast<uint32>(pagesize) > 0);
+      static_assert(pagesize > 0, "");
```

Comment:
Could you please add `page size of zero is not allowed`.

Reply:
No, because I just removed the static_assert as by your previous comment ;)
To further elaborate on not having a message: I am very used to C++17, where the message is optional and I usually do not provide one if the checked condition is simple. Furthermore, there are also no messages attached to runtime assert(...)s in the code.

Reply:
Upsi.